src/biomechanic/optimize_bend.jl

```diff
@@ -0,0 +1,279 @@
+"""
+    optimize_bend(field_data,
+                elastic_modulus,
+                shear_modulus,
+                type,
+                method,
+                tol,
+                nb_rep,
+                kwargs...)
+
+Optimize the elastic_modulus, the shear_modulus or both using observations.
+
+# Arguments
+
+- `field_data`: DataFrames, or list of DataFrames with the x, y and z coordinates of the observed points.
+If list of DataFrames, each element correspond to a separate beam (e.g. each branch)
+- `elastic_modulus`: Elasticity modulus (bending, MPa). Default is (1.0,10000.0).
+See details.
+- `shear_modulus`: Shear modulus (torsion, MPa). Default is (1.0,10000.0).
+`See details.
+- `type`: Type of optimization (either "bending","torsion" or "all"). Default is "all".
+- `method`: Method to use when optimizing one parameter. Default is "L-BFGS-B".
+See details.
+- `tol`: Tolerance for optimization accuracy. Default is `eps()^0.25`.
+- `nb_rep`: Number of starting points for the optimization algorithm. Default is 5.
+- `kwargs...`: Further parameters to pass to `bend()`
+
+# Details
+
+The `elastic_modulus` and `shear_modulus` are either provided as a vector of min and max values if optimized,
+or as a single value to force the parameter if not optimized.
+
+# Returns
+
+A NamedTuple with the following elements:
+- `elastic_modulus`: the optimized elasticity modulus value
+- `shear_modulus`: the optimized shear modulus value
+- `init_values`: a DataFrames with the initial values used for each repetition
+- `optim_values`: a DataFrames with the optimized values for each step
+- `min_quadratic_error`: minimum quadratic error of all repetitions
+- `rep_min_crit`: index of the repetition that gave the minimum quadratic error
+- `plots`: plots of optimal value ~ initial value for each parameter to analyze the sensitivity of the optimized value to the starting points.
+
+# Examples
+```julia
+file = joinpath(dirname(dirname(pathof(VPalm))), "test", "files", "6_EW01.22_17_kanan_unbent.csv")
+fd = DataFrame(CSV.File(file))
+
+```
+"""
+function optimize_bend(field_data;
+    elastic_modulus=(1.0, 10000.0),
+    shear_modulus=(1.0, 10000.0),
+    type="all",
+    method="L-BFGS-B",
+    tol=eps()^0.25,
+    nb_rep=5,
+    kwargs...)
+
+    # Check optimization type
+    type ∈ ["all", "bending", "torsion"] || error("Must be either 'all', 'bending' or 'torsion'")
+
+    # Required columns
+    required_columns = ["x","y","z","distance",
+         "type","width","height","inclination",
+         "torsion","mass","mass_right","mass_left"]
+
+    # Convert to dict if necessary
+    field_data = field_data isa DataFrame ? Dict("sim1" => field_data) : field_data
```

Comment:
It would be better to ditch the DataFrame dependency and provide the data as a Tables.jl-compatible data format instead. This way we can pass the data in any format, but above all we don't depend on DataFrames, only on Tables.jl that is very stable and lightweight.